synapse/logging/_terse_json.py

```diff
@@ -171,7 +171,7 @@ def pauseProducing(self):
 
     def stopProducing(self):
         self._paused = True
-        self._buffer = None
+        self._buffer = deque()
```

Comment:
`.clear`, maybe?

Reply:
For things that don't get reused I tend to bias to creating new objects simply to help ensure the underlying resources gets GC'ed, though that might be a bit spurious

Reply:
hrm. I'd be more concerned about the overhead of creating new objects.

I think the whole thing's academic here.